crates/CrateAssembler.kt

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright (C) 2021 Vaticle
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import com.eclipsesource.json.Json
+import com.eclipsesource.json.JsonArray
+import com.eclipsesource.json.JsonObject
+import com.fasterxml.jackson.dataformat.toml.TomlMapper
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry
+import org.apache.commons.compress.archivers.tar.TarArchiveOutputStream
+import org.apache.commons.compress.utils.IOUtils
+import picocli.CommandLine
+import picocli.CommandLine.Command
+import picocli.CommandLine.Option
+import java.io.BufferedOutputStream
+import java.io.ByteArrayInputStream
+import java.io.File
+import java.nio.charset.StandardCharsets
+import java.nio.file.Path
+import java.util.concurrent.Callable
+import java.util.zip.GZIPOutputStream
+import kotlin.system.exitProcess
+
+
+data class CargoTomlPackage(
+    val name: String,
+    val edition: String,
+    val version: String,
+    val authors: List<String>,
+    val homepage: String,
+    val repository: String,
+    val documentation: String,
+    val description: String,
+    val readme: String
+)
+
+data class CargoToml(
+    val `package`: CargoTomlPackage,
+    val dependencies: Map<String, String>
+)
+
+
+@Command(name = "crate-assembler", mixinStandardHelpOptions = true)
+class CrateAssembler : Callable<Unit> {
+    @Option(names = ["--srcs"], split = ";", required = true)
```

Comment:
I think the `--srcs` and `--deps` will not split correctly when receiving an empty list. Let's fix it by taking inspiration from how Joshua fixed `assemble_maven`: https://github.com/vaticle/bazel-distribution/pull/320.

Reply:
@lolski addressed in https://github.com/vaticle/bazel-distribution/pull/322/commits/f724b1baade55d6e79c2e908c094d29bf625180e and https://github.com/vaticle/bazel-distribution/pull/322/commits/cc973c0b1c515d37716cd2b6f6efb1578d36babe